### PR TITLE
Disallow await as bound name in using declaration

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Implements https://github.com/tc39/proposal-explicit-resource-management/pull/138
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

This PR starts by simplifying let-as-name binding checking, but then I realize that the `using` declaration has already banned binding patterns, which greatly simplifies the await-checking. Anyway I am pleased that we get rid of a parameter from the hot path `checkIdentifier`.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15391"></a>

### CODE REVIEW

File: packages/babel-parser/test/fixtures/experimental/explicit-resource-management/invalid-using-binding-await/options.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "sourceType": "script"
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/test/fixtures/experimental/explicit-resource-management/invalid-using-binding-await/options.json:2]: Can you also add a test in modules? It throws that `await` is reserved in modules, right?

File: packages/babel-parser/test/fixtures/experimental/explicit-resource-management/invalid-using-binding-await-module/output.json [hunk 2]
```diff
@@ -0,0 +1,166 @@
+{
+  "type": "File",
+  "start":0,"end":113,"loc":{"start":{"line":1,"column":0,"index":0},"end":{"line":12,"column":1,"index":113}},
+  "errors": [
+    "SyntaxError: Can not use 'await' as identifier inside an async function. (2:8)",
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/test/fixtures/experimental/explicit-resource-management/invalid-using-binding-await-module/output.json:5]: Unrelated, but this error message is wrong (we are not in an async function, but at the top-level of a module).

**Review by nicolo-ribaudo — APPROVED** [review]

File: packages/babel-parser/test/fixtures/experimental/explicit-resource-management/invalid-using-binding-await-module/output.json (hunk 2 above)
Comment (JLHwung) [reply packages/babel-parser/test/fixtures/experimental/explicit-resource-management/invalid-using-binding-await-module/output.json:5]: We don't differentiate between module top level and async function as long as `[Await]` production parameter set. Maybe rephrase it as "Can not use 'await' as identifier inside an async context"?

**Review by liuxingbaoyu — APPROVED** [review]
